generators/gae/index.js

```diff
@@ -141,8 +143,91 @@ module.exports = class extends BaseGenerator {
+                        this.debug('Error:', err);
+                    }
+                }
+            }
+        });
+
+        return appsFolders;
+    }
+
     get prompting() {
         return {
+            askForPath() {
+                if (this.abort) return undefined;
+                if (this.applicationType !== 'gateway') return undefined;
+                const messageAskForPath = 'Enter the root directory where the microservices are located';
+                const prompts = [
+                    {
+                        type: 'input',
+                        name: 'directoryPath',
+                        message: messageAskForPath,
+                        default: this.directoryPath || '../',
+                        validate: input => {
+                            const path = this.destinationPath(input);
+                            if (shelljs.test('-d', path)) {
+                                const appsFolders = this.getMicroserviceFolders(input);
+
+                                if (appsFolders.length === 0) {
+                                    return `No microservices are found in ${path}`;
+                                }
+                                return true;
+                            }
+                            return `${path} is not a directory or doesn't exist`;
+                        },
+                    },
+                ];
+
+                return this.prompt(prompts).then(props => {
+                    this.directoryPath = props.directoryPath;
+                    // Patch the path if there is no trailing "/"
+                    if (!this.directoryPath.endsWith('/')) {
+                        this.log(chalk.yellow(`The path "${this.directoryPath}" does not end with a trailing "/", adding it anyway.`));
+                        this.directoryPath += '/';
+                    }
+                    this.appsFolders = this.getMicroserviceFolders(this.directoryPath);
+                    this.log(chalk.green(`${this.appsFolders.length} applications found at ${this.destinationPath(this.directoryPath)}\n`));
+                });
+            },
+            askForApps() {
+                if (this.regenerate) return;
+                if (this.applicationType !== 'gateway') return;
+                const done = this.async();
+                const messageAskForApps = 'Which microservice applications do you want to include in your configuration?';
+                const prompts = [
+                    {
+                        type: 'checkbox',
+                        name: 'chosenApps',
+                        message: messageAskForApps,
+                        choices: this.appsFolders,
+                        default: this.defaultAppsFolders,
+                        validate: input => (input.length === 0 ? 'Please choose at least one application' : true),
+                    },
+                ];
+
+                this.prompt(prompts).then(props => {
```

Comment:
Return prompt here too.

Reply:
Thanks much for the explanation. Done. :+1: